profiler/src/ProfilerEngine/Datadog.Profiler.Native.Windows/OsSpecificApi.cpp

```diff
@@ -152,4 +155,94 @@ bool IsRunning(ManagedThreadInfo* pThreadInfo, uint64_t& cpuTime)
     return IsRunning(sti.ThreadState);
 }
 
+
+bool InitializeGetLogicalProcessorInformationCallback()
+{
+    auto hModule = GetModuleHandleA("kernel32.dll");
+    if (hModule == nullptr)
+    {
+        Log::Error("Impossible to load kernel32.dll: 0x", std::hex, GetLastError());
+        return false;
+    }
+
+    GetLogicalProcessorInformation = (GetLogicalProcessorInformation_)GetProcAddress(hModule, "GetLogicalProcessorInformation");
+    if (GetLogicalProcessorInformation == nullptr)
+    {
+        Log::Error("Impossible to get GetLogicalProcessorInformation: 0x", std::hex, GetLastError());
+        return false;
+    }
+
+    return true;
+}
+
+// from https://learn.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getlogicalprocessorinformation
+//
+DWORD CountSetBits(ULONG_PTR bitMask)
+{
+    DWORD LSHIFT = sizeof(ULONG_PTR) * 8 - 1;
+    DWORD bitSetCount = 0;
+    ULONG_PTR bitTest = (ULONG_PTR)1 << LSHIFT;
+    DWORD i;
+
+    for (i = 0; i <= LSHIFT; ++i)
+    {
+        bitSetCount += ((bitMask & bitTest) ? 1 : 0);
+        bitTest /= 2;
+    }
+
+    return bitSetCount;
+}
+
+bool GetProcessorCount(uint32_t& processorCount)
+{
+    if (GetLogicalProcessorInformation == nullptr)
+    {
+        return false;
+    }
+
+    // TODO: use std::unique_ptr
+    PSYSTEM_LOGICAL_PROCESSOR_INFORMATION buffer = NULL;
+    PSYSTEM_LOGICAL_PROCESSOR_INFORMATION ptr = NULL;
+    DWORD returnLength = 0;
+    DWORD byteOffset = 0;
+
+    auto result = GetLogicalProcessorInformation(buffer, &returnLength);
+    if (result == 0)
+    {
+        if (GetLastError() != ERROR_INSUFFICIENT_BUFFER)
+        {
+            return false;
+        }
+
+        buffer = (PSYSTEM_LOGICAL_PROCESSOR_INFORMATION)malloc(returnLength);
+        if (nullptr == buffer)
+        {
+            return false;
+        }
+
+        result = GetLogicalProcessorInformation(buffer, &returnLength);
+        if (result == 0)
+        {
+            free(buffer);
+            return false;
+        }
+    }
+
+    ptr = buffer;
+
+    while (byteOffset + sizeof(SYSTEM_LOGICAL_PROCESSOR_INFORMATION) <= returnLength)
+    {
+        if (ptr->Relationship == RelationProcessorCore)
+        {
+            // A hyperthreaded core supplies more than one logical processor.
+            processorCount += CountSetBits(ptr->ProcessorMask);
+        }
+
+        byteOffset += sizeof(SYSTEM_LOGICAL_PROCESSOR_INFORMATION);
+        ptr++;
+    }
+
```

Comment:
oops missing `free` at the end
But if you go for a `std::unique_ptr`, you won't have to worry ;)